src/node_http_parser.cc

```diff
@@ -644,6 +733,26 @@ class Parser : public AsyncWrap, public StreamListener {
     args.GetReturnValue().Set(ret);
   }
 
+  static void Duration(const FunctionCallbackInfo<Value>& args) {
+    Parser* parser;
+    ASSIGN_OR_RETURN_UNWRAP(&parser, args.Holder());
+
+    if (parser->last_message_start_ == 0) {
+      args.GetReturnValue().Set(0);
+      return;
+    }
+
+    double duration = (uv_hrtime() - parser->last_message_start_) / 1000000;
```

Comment:
```suggestion
    double duration = (uv_hrtime() - parser->last_message_start_) / 1e6;
```
Otherwise, the result would get truncated due to integer division

Reply:
👍